tests/Query/BuilderTest.php

```diff
@@ -433,6 +434,62 @@ function (Builder $builder) {
                 ->orWhereNotBetween('id', collect([3, 4])),
         ];
 
+        yield 'where like' => [
+            ['find' => [['name' => new Regex('^acme$', '')], []]],
+            fn (Builder $builder) => $builder->where('name', 'like', 'acme'),
+        ];
+
+        yield 'where like escape' => [
+            ['find' => [['name' => new Regex('^\^acme\$$', '')], []]],
+            fn (Builder $builder) => $builder->where('name', 'like', '^acme$'),
+        ];
+
+        yield 'where like %' => [
+            ['find' => [['name' => new Regex('.*acme.*', '')], []]],
+            fn (Builder $builder) => $builder->where('name', 'like', '%acme%'),
+        ];
+
+        yield 'where ilike' => [
+            ['find' => [['name' => new Regex('^acme$', 'i')], []]],
+            fn (Builder $builder) => $builder->where('name', 'ilike', 'acme'),
+        ];
+
+        yield 'where not like' => [
+            ['find' => [['name' => ['$not' => new Regex('^acme$', '')]], []]],
+            fn (Builder $builder) => $builder->where('name', 'not like', 'acme'),
+        ];
+
+        yield 'where not ilike' => [
+            ['find' => [['name' => ['$not' => new Regex('^acme$', 'i')]], []]],
+            fn (Builder $builder) => $builder->where('name', 'not ilike', 'acme'),
+        ];
+
+        $regex = new Regex('^acme$', 'si');
+        yield 'where BSON\Regex' => [
+            ['find' => [['name' => $regex], []]],
+            fn (Builder $builder) => $builder->where('name', 'regex', $regex),
+        ];
+
+        yield 'where regex' => [
+            ['find' => [['name' => $regex], []]],
+            fn (Builder $builder) => $builder->where('name', 'regex', '/^acme$/si'),
+        ];
+
+        yield 'where not regex' => [
+            ['find' => [['name' => ['$not' => $regex]], []]],
+            fn (Builder $builder) => $builder->where('name', 'not regex', '/^acme$/si'),
+        ];
+
+        yield 'where regexp' => [
+            ['find' => [['name' => $regex], []]],
+            fn (Builder $builder) => $builder->where('name', 'regexp', '/^acme$/si'),
+        ];
+
+        yield 'where not regexp' => [
+            ['find' => [['name' => ['$not' => $regex]], []]],
+            fn (Builder $builder) => $builder->where('name', 'not regexp', '/^acme$/si'),
```

Comment:
Should we have a test with different delimiter characters, or does `regex` and its `regexp` alias require that the pattern be wrapped with `/`?

Reply:
Before this PR, only `/` is supported. Since this is the delimiter of regex in MongoDB, I think it's fine to stick with it. It could be easy to support other delimiters but is that better?

Reply:
> Since this is the delimiter of regex in MongoDB

The delimiter technically has more to do with the JavaScript REPL than the server. From the server's perspective, there are only two ways to express a regular express:

 * BSON regex type, which consists of two separate null-terminated strings for the pattern and options, respectively.
 * [$regex](https://www.mongodb.com/docs/manual/reference/operator/query/regex/) operator, which accepts either a string with no delimiters or a BSON regex object as its value. A separate string is used to solicit options via an `$options` field unless those are already embedded in the BSON regex object passed to `$regex`.

When writing that comment, I was thinking more of PHP's own [delimiter](https://www.php.net/manual/en/regexp.reference.delimiters.php) syntax. There's a lot there that isn't relevant, but being able to use something like `#` instead of `/` in a pattern that might need to match slashes and avoid escaping seems like a valid use case. I'm not convinced we need to offer this functionality, though.

Feel free to resolve after reading.

Reply:
This should be implemented in [PHPLIB-1207](https://jira.mongodb.org/browse/PHPLIB-1207).